src/aiortc/contrib/media.py

```diff
@@ -418,18 +458,71 @@ async def __run_track(self, track, context):
             try:
                 frame = await track.recv()
             except MediaStreamError:
-                print("Couldn't receive the track!")
+                print("Couldn't receive the {} track!".format(track.kind))
                 return
-            if track.kind != "keypoints":
+
+            if track.kind == "video":
+                if enable_prediction:
+                    # update model related info with most recent frame
+                    self.received_video_frame += 1
+                    self.__log_debug("Received video frame %s: %s", self.received_video_frame, frame)
+                    if self.received_video_frame % UPDATE_SRC_FREQ == 0:
+                        #TODO how to synchronize this with received keypoints
```

Comment:
I think we should change the logic here to use any received video frame as a source frame (otherwise really wasteful to even receive frames) and not send frames at the sender unless it is meant to be used as a source frame. Don't need the receiver to do this check at all then and seems cleaner.

Reply:
done!